apps/common/behaviours/casework-submission.js

```diff
@@ -6,14 +6,20 @@ const StatsD = require('hot-shots');
 const client = new StatsD();
 
 const Compose = func => superclass => class extends superclass {
-  prepare() {
-    if (typeof func === 'function') {
-      const model = new AuthToken();
-      return model.auth().then(token => {
-        return Object.assign(super.prepare(token), func(this.toJSON(), token));
-      });
+  async prepare() {
+    try {
+      if (typeof func === 'function') {
+        const model = new AuthToken();
+        const token = await model.auth();
+        const prepareToken = await super.prepare(token);
+        return Object.assign(prepareToken, func(this.toJSON(), token));
+      }
+      return await super.prepare();
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error('Error in prepare method:', e);
```

Comment:
hof logger 😬 
you've already done the work so if you don't have time I understand

Reply:
I hsve added in the logger now